### PR TITLE
Rename AudioStreamSample to a more discoverable name

### LINKED ISSUE #4532: Replace the "Sample" from AudioStreamSample's name with something else
### Describe the project you are working on

A game that uses lots of sound effects.

### Describe the problem or limitation you are having in your project

When I first saw "AudioStreamSample" I thought it was a resource that imported any kind of audio file, while what it actually imported was just WAV format. In fact, I later found out AudioStreamSample isn't just for WAVs, but a resource that stores PCM data, allowing it to be used to store generated data or microphone recordings.

I think "Sample" might be misleading in some cases.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Simply rename the class to something else, like AudioStreamWAV or AudioStreamPCM. PCM would be more specific.

This was ***sort of*** discussed in a pull request, and ellenhp asked to do it as a feature proposal. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

All references to "AudioStreamSample" would change, simply.

There are already two pull requests ready for the options mentioned above:

[AudioStreamWAV](https://github.com/godotengine/godot/pull/60953) (there's already some discussion there)

[AudioStreamPCM](https://github.com/godotengine/godot/pull/60957)

Approve any of those, and this proposal is accepted.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

While this isn't exactly "core", you can't rename a class with an addon.

### PR DESCRIPTION
**Pull request by DeeJayLSP**
Supersedes #60953 and implements godotengine/godot-proposals#4532, which is now approved.

_Closes https://github.com/godotengine/godot-proposals/issues/4532_

_Applies https://github.com/godotengine/godot-proposals/issues/4532_

### CODE REVIEW

**Comment by DeeJayLSP** [comment]
Marked as Draft for discussion, please check godotengine/godot-proposals#4532

**Comment by DeeJayLSP** [comment]
This is now ready

File: editor/project_converter_3_to_4.cpp [hunk 1]
```diff
@@ -120,7 +120,7 @@ static const char *enum_renames[][2] = {
 	{ "JOINT_PIN", "JOINT_TYPE_PIN" }, // PhysicsServer2D
 	{ "JOINT_SLIDER", "JOINT_TYPE_SLIDER" }, // PhysicsServer3D
 	{ "KEY_CONTROL", "KEY_CTRL" }, // Globals
-	{ "LOOP_PING_PONG", "LOOP_PINGPONG" }, //AudioStreamSample
+	{ "LOOP_PING_PONG", "LOOP_PINGPONG" }, //AudioStreamPCM
```
Comment (nathanfranke) [inline editor/project_converter_3_to_4.cpp:123]: I am not familiar with the project converter, but is there actually some code that converts "AudioStreamSample" to "AudioStreamPCM"? I just see a comment change here.

Comment (DeeJayLSP) [reply editor/project_converter_3_to_4.cpp:123]: I wasn't aware of the class renames, I added that now.

**Comment by DeeJayLSP** [comment]
Redone in order to fix conflicts.

I also realized I forgot to rename AudioStreamPlaybackSample to AudioStreamPlaybackPCM.

**Comment by DeeJayLSP** [comment]
@YuriSizov This has been ready like 2 weeks ago, and your tweet kinda made me anxious. This is just me attempting to get in touch 😕 sorry for the inconvenience.

**Comment by ellenhp** [comment]
I don't think it was Yuri's fault that this didn't get looked at, that's just me being too stressed with a job search to work on Godot. :) Looks fine to me if Nathan's comment was resolved. Does anyone have objections (keeping in mind that the proposal was approved)?

**Comment by DeeJayLSP** [comment]
I just mentioned Yuri because of the tweet but again, sorry for the inconvenience.

**Comment by ellenhp** [comment]
Makes sense. I'm going to approve this for the audio team but I'll let someone else decide whether that's sufficient to merge.

**Review by ellenhp — APPROVED** [review]
LGTM for audio changes.

**Comment by akien-mga** [comment]
@DeeJayLSP As mentioned in the blog post, the feature freeze on August 3 is for **new PRs**. This one is already open so there's no worry to have.

The change itself is trivial, what's pending here is to decide if we actually want to do this rename.

**Comment by YuriSizov** [comment]
We've discussed it again in a review meeting and this time agreed that calling it WAV would be more discoverable for the end user. Sorry for the inconvenience, can you re-rename it?

**Comment by DeeJayLSP** [comment]
I am not willing to reopen the pull request that precedes this one (in fact, it's extremely outdated) so I'm going to work on this one.

**Comment by DouglasNPD** [comment]
Accidentally commited using my work account

**Comment by DeeJayLSP** [comment]
Accidentally commited using my work account '-'

File: editor/project_converter_3_to_4.cpp [hunk 2]
```diff
@@ -120,7 +120,7 @@ static const char *enum_renames[][2] = {
 	{ "JOINT_PIN", "JOINT_TYPE_PIN" }, // PhysicsServer2D
 	{ "JOINT_SLIDER", "JOINT_TYPE_SLIDER" }, // PhysicsServer3D
 	{ "KEY_CONTROL", "KEY_CTRL" }, // Globals
-	{ "LOOP_PING_PONG", "LOOP_PINGPONG" }, //AudioStreamSample
+	{ "LOOP_PING_PONG", "LOOP_PINGPONG" }, //AudioStreamWAV
```
Comment (akien-mga) [inline editor/project_converter_3_to_4.cpp:123]: Nitpick but while at it:
```suggestion
	{ "LOOP_PING_PONG", "LOOP_PINGPONG" }, // AudioStreamWAV
```

File: editor/import/resource_importer_wav.cpp [hunk 3]
```diff
@@ -33,7 +33,7 @@
 #include "core/io/file_access.h"
 #include "core/io/marshalls.h"
 #include "core/io/resource_saver.h"
-#include "scene/resources/audio_stream_sample.h"
+#include "scene/resources/audio_stream_pcm.h"
```
Comment (akien-mga) [inline editor/import/resource_importer_wav.cpp:36]: Missing that one.

**Review by akien-mga — APPROVED** [review]
Good to merge aside from one wrong include.

**Comment by DeeJayLSP** [comment]
image
The icon for it doesn't seem to work despite having the file renamed.

**Comment by Calinou** [comment]
>  The icon for it doesn't seem to work despite having the file renamed.

Did you recompile the editor *after* renaming the icon?

**Comment by DeeJayLSP** [comment]
> > The icon for it doesn't seem to work despite having the file renamed.
> 
> Did you recompile the editor _after_ renaming the icon?

I downloaded the build artifact from Actions and ran it.

**Comment by akien-mga** [comment]
Tested locally and it works, might be a cache issue.

**Comment by akien-mga** [comment]
Thanks!

**Comment by DeeJayLSP** [comment]
How ironic.

Everything started as a PR that renamed it to AudioStreamWAV.

Then, it ended in a PR that renamed it to AudioStreamWAV.

**Comment by akien-mga** [comment]
Hehe yeah, that's why this kind of change is best evaluated on the proposal than on multiple alternative PRs. When the change is trivial to make but up for bikeshedding, what needs to come first is the decision, not the ready-to-merge PR.

**Comment by tinmanjuggernaut** [comment]
Why was OGG recased to Ogg #62164, when WAV was not? Ogg is the official name, and Wave is the official name.

AudioStreamOggVorbis names both a container (ogg) and a codec (vorbis) with official names. Yet AudioStreamWAV only has a nickname for the container (wav) and not the codec (PCM). PCM is an acronym so should be capitals, like AudioStreamMP3. But WAV is not an acronym.

This makes the API look sloppy. Sample wasn't good so it's good to update these names. However, for consistency and correctness, AudioStreamOggVorbis should have been named AudioStreamOgg, and AudioStreamWAV should be renamed AudioStreamWave. Vorbis doesn't belong, and WAV is not the correct name.

Separate things:
Ogg
https://www.xiph.org/ogg/

Vorbis
https://xiph.org/vorbis/

**Comment by YuriSizov** [comment]
The name `AudioStreamWAV` was picked after a lot of back and forth because it makes the class easier to discover to a layperson who is unlikely to know beyond the file extension. `PCM` is not something most people are familiar with.

**Comment by YuriSizov** [comment]
Well, that's my point. It was renamed to WAV because of the familiarity, and consistency wasn't the goal. Wave can also be confusing, just as PCM, though likely less so.

**Pull request merged**